### PR TITLE
Update the AnimStateGraph class documentation

### LINKED ISSUE #3674: AnimStateGraph constructor is documented as having zero arguments
https://github.com/playcanvas/engine/blob/dev/src/anim/state-graph/anim-state-graph.js#L1-L7

If we intend for this to be public API, we should document the argument that it takes.

### PR DESCRIPTION
**Pull request by ellthompson**
The AnimStateGraph class was documented as having no arguments in it's constructor. This PR hides the constructor from the jsdocs and gives an example of how to instantiate the class.

Fixes #3674 

I confirm I have read the [contributing guidelines](https://github.com/playcanvas/engine/blob/master/.github/CONTRIBUTING.md) and signed the [Contributor License Agreement](https://docs.google.com/a/playcanvas.com/forms/d/1Ih69zQfJG-QDLIEpHr6CsaAs6fPORNOVnMv5nuo0cjk/viewform).

### CODE REVIEW

**Comment by mvaligursky** [comment]
Do we need to expose the constructor at all?
In the engine examples, the state graph is created from json data this way:
`        modelEntity.anim.loadStateGraph(animStateGraphData);`

Do we need to expose this other way as well?
Perhaps we do, just making sure it's something users would use.

**Comment by ellthompson** [comment]
Editor users that load their state graphs in scripts will get an instance of this class as the asset resource. I was thinking of keeping it public for that specific use case. I'd still want to be moving engine only users away from interacting with this class though.

**Comment by yaustar** [comment]
A little off topic, maybe we should flag functions that are used by the Editor but aren't public API to help with things like this?

**Comment by mvaligursky** [comment]
It's ok to keep the class public for the reason as mentioned, but do we need to have the constructor public? That's what I was asking. You could use @hideconstructor on it to make constructor hidden.

**Comment by ellthompson** [comment]
I didn't know about that jsdoc property. I'll add that, remove the @param property and update the @example to be more editor specific.

**Review by mvaligursky — APPROVED** [review]
looks good. please update the PR description to match the final changes.

File: src/anim/state-graph/anim-state-graph.js [hunk 1]
```diff
@@ -1,6 +1,10 @@
 /**
- * Creates an AnimStateGraph asset resource from a blob of JSON data that represents an anim state
- * graph.
+ * Creates an AnimStateGraph asset resource from a blob of JSON data that represents an anim state graph. Can be loaded into an anim component using the {@link AnimComponent#loadStateGraph} method
+ *
+ * @hideconstructor
+ * @example
+ * const animStateGraph = asset.resource;
+ * entity.anim.loadStateGraph(animStateGraph);
```
Comment (willeastcott) [inline src/anim/state-graph/anim-state-graph.js:7]: This should go on the constructor JSDoc block.

**Pull request merged**